### PR TITLE
Add Pocket Casts embed block variation

### PR DESCRIPTION
**Pull request by phcp**
## What?

Introduces the embed block variation for Pocket Casts ([WP55860](https://core.trac.wordpress.org/ticket/55860)).

## Why?

To enable embedding Pocket Casts on WordPress posts using blocks.

## How?

Adding the variation attributes for the Pocket Casts in the [embed block variations array](https://github.com/WordPress/gutenberg/pull/41006/files#diff-0fdea61a997957a9055aed5beb352cda8b21d423e872819452f36c185307251bR220), also adding the Pocket Casts icon on the [embed icons file](https://github.com/WordPress/gutenberg/pull/41006/files#diff-7249e9bec4eb469b312fabfdcd6d34041ac16e808697d9e08706591402304ba3R157).

#### By adding the Pocket Casts block
1. Create an empty post
2. Add the Pocket Casts block
3. Enter a Pocket Casts URL
3.1. Examples: `https://pca.st/k2rof194`, `https://pca.st/qekyjgs6`, `https://pca.st/63wN`
5. Click on Embed
6. Check if the Pocket Casts player is displayed

#### By pasting a Pocket Casts URL 
1. Create an empty post
2. Paste a Pocket Casts URL in the content editor
2.1. Examples: `https://pca.st/k2rof194`, `https://pca.st/qekyjgs6`, `https://pca.st/63wN`
4. Check if the URL is replaced by the Pocket Casts player

## Screenshots

Screen Shot 2022-05-30 at 14 04 30
Screen Shot 2022-05-30 at 14 10 51
Screen Shot 2022-05-30 at 12 31 13

### CODE REVIEW

File: packages/block-library/src/embed/variations.js [hunk 1]
```diff
@@ -216,6 +217,15 @@ const variations = [
 		patterns: [ /^https?:\/\/(www\.)?mixcloud\.com\/.+/i ],
 		attributes: { providerNameSlug: 'mixcloud', responsive: true },
 	},
+	{
+		name: 'pocketcasts',
+		title: 'Pocket Casts',
+		icon: embedPocketCastsIcon,
+		keywords: [ __( 'pocketcasts' ), __( 'podcast' ), __( 'audio' ) ],
```
Comment (mtias) [inline packages/block-library/src/embed/variations.js]: this should be unnecessary if it's in the name

Comment (phcp) [reply packages/block-library/src/embed/variations.js]: Fixed!

**Review by mtias — APPROVED** [review]

**Comment by peterwilsoncc** [comment]
The WP trac ticket was originally closed due to auto-discovery working.

I've reopened it pending the inclusion of a block ([WP#55860 (comment)](https://core.trac.wordpress.org/ticket/55860#comment:3)). Once this PR is merged I will add the trac ticket to the next milestone.

**Comment by desrosj** [comment]
@phcp While collecting props and credits for 6.1, it seems you are one of a few dozen folks that have not [connected their WordPress.org account to their GitHub profile](https://make.wordpress.org/core/2020/03/19/associating-github-accounts-with-wordpress-org-profiles/).

Could you share your WordPress.org account so that we can ensure you're properly credited on the About page for WordPress 6.1? [Connecting your accounts ](https://make.wordpress.org/core/2020/03/19/associating-github-accounts-with-wordpress-org-profiles/)will also help those collecting this in the future. Thanks in advance!

**Comment by phcp** [comment]
@desrosj My WordPress.org account is `paulohcruz`, and now it's connected with my GH account. Thank you!

**Comment by wayneconnor** [comment]
""need to subscribe to a premium Pocket Casts account to add your favorite or own Pocket Casts content into the block"

Am I reading this correctly?

**Pull request merged**